docs/deeplinking.md

```diff
@@ -0,0 +1,403 @@
+        deeplink: deeplink,
+      );
+}
+```
+
+- features/deeplink/presentation/deeplink_wm.dart
+
+```dart
+/// DI factory for [DeeplinkWM].
+DeeplinkWM defaultDeeplinksWMFactory(BuildContext context) {
+  final deeplinkScope = context.read<IDeeplinkScope>();
+
+  return DeeplinkWM(
+    DeeplinkModel(
+      deeplinkRepository: deeplinkScope.repository,
+      deeplinkOpener: deeplinkScope.deeplinkOpener,
+    ),
+  );
+}
+
+/// [DeeplinkWM] interface.
+abstract interface class IDeeplinkWM with ILocalizationMixin, ThemeIModelMixin implements IWidgetModel {}
+
+/// [WidgetModel] for Deeplink features.
+final class DeeplinkWM extends WidgetModel<DeeplinkWidget, DeeplinkModel>
+    with LocalizationMixin, ThemeWMMixin
+    implements IDeeplinkWM {
+  late final StreamSubscription<Uri> _deeplinksSub;
+
+  /// Default constructor for [DeeplinkWM].
+  DeeplinkWM(
+    super._model,
+  );
+
+  @override
+  void initWidgetModel() {
+    super.initWidgetModel();
+
+    _deeplinksSub = model.onDeeplink.listen(_onDeeplink);
+    model.initialDeeplink.then(_handleInitialDeeplink);
+  }
+
+  void _handleInitialDeeplink(Uri? deeplink) {
+    if (deeplink != null) model.openDeeplink(deeplink);
+  }
+
+  void _onDeeplink(Uri deeplink) => model.openDeeplink(deeplink);
+
+  @override
+  void dispose() {
+    _deeplinksSub.cancel();
+
+    super.dispose();
+  }
+}
+```
+
+- features/deeplink/presentation/deeplink_widget.dart
+
+```dart
+/// [ElementaryWidget] for Deeplink features.
+class DeeplinkWidget extends ElementaryWidget<IDeeplinkWM> {
+  /// Child [Widget]
+  final Widget child;
+
+  /// Default constructor for [DeeplinkWidget].
+  const DeeplinkWidget({
+    required this.child,
+    super.key,
+  }) : super(defaultDeeplinksWMFactory);
+
+  @override
+  Widget build(IDeeplinkWM wm) {
+    return child;
+  }
+}
+```
+
+## Inject deeplink feature to app
+
+- runner.dart
+
+```dart
+  // ...
+
+  final initialDeeplink = await scope.deeplinkRepository.initialDeeplink;
+  
+  // ...
+
+  runApp(
+    AppFlow(
+      // ...
+      initialDeeplink: initialDeeplink,
+      // ...
+    ),
+  );
+
+  // ...
+```
+
+- features/app/app.dart
+
+Update MaterialApp
+
+```dart
+    // ..
+
+        return MaterialApp.router(
+          // ...
+          builder: (context, child) => DeeplinkFlow(
+            child: child,
+          ),
+          // ...
+        );
+    // ...
+```
+
+- features/app/app_flow.dart
+
+Add initialDeeplink field
+
+```dart
+  // ...
+
+  /// Initial deeplink that launches the app.
+  final Uri? initialDeeplink;
+  
+  // ...
+```
+
+Update build method
+
+```dart
+  // ...
+
+  @override
+  Widget build(BuildContext context) {
+    return DiScope<IAppScope>(
+      factory: (_) => appScope,
+      child: Nested(
+        children: [
+          ChangeNotifierProvider<AppRouter>(create: (_) => AppRouter()),
+          const ThemeModeProvider(),
+        ],
+        child: const App(),
+      ),
+    );
+  }
```

Comment:
I don't see a reason why should we modify the code this way

Reply:
I faced problems providing everything in Nested, I'll check again if error still exist and return with answer

Reply:
I don't see any problems here. Yes, can you doublecheck it, please?